lib/command.js

```diff
@@ -814,6 +814,25 @@ Expecting one of '${allowedValues.join("', '")}'`);
     return this._optionValueSources[key];
   }
 
+  /**
+    * Get source of option value. See also .optsWithGlobals().
+    * Expected values are default | config | env | cli | implied
+    *
+    * @param {string} key
+    * @return {string}
+    */
+
+  getOptionValueSourceWithGlobals(key) {
+    // global overwrites local, like optsWithGlobals
+    let source;
```

Comment:
For reference :
You can also write the following using `findLast`.

```js
    return getCommandAndParents(this)
      .findLast((cmd) => cmd.getOptionValueSource(key) !== undefined)
      .getOptionValueSource(key);
```

Reply:
You can use Node.js 18 or later.
My apologies.

Reply:
I started out planning to target Commander v10.x just so I could use `??`, but switched to old-school Node 12 compatible code for now. Eventually!